### DISCUSSION TITLE
Please add Controlnet for Flux models

### DISCUSSION THREAD

**Discussion by ivaxsirc** [lllyasviel/stable-diffusion-webui-forge#1299] [General]
Flux in Forge without controlnet it's like a cake missing an essential ingredient

Thanks

**Comment by Deonix59** [comment]
What's also missing is that the weights in the promt don't work.

**Reply by jajajajajajajajajajajajajajajajajajajaj** [reply]
that only works if you cntrl+up arrow key  AND add brackets. it doesnt register the number you enter from the cntrl+up but it does register the number of brackets that you choose to use. you still need to do the cntrl+up otherwise it wont register the brackets at all

**Comment by wzwow** [comment]
please please please add controlnet for flux

**Comment by nitinmukesh** [comment]
And these are the available controlnet for FLUX that I know

Xlabs v3 controlnets: https://huggingface.co/XLabs-AI/flux-controlnet-collections
InstantX + Shakkerlabs union pro controlnet: https://huggingface.co/Shakker-Labs/FLUX.1-dev-ControlNet-Union-Pro
Mistoline: https://huggingface.co/TheMistoAI/MistoLine_Flux.dev

**Reply by ZeroCool22** [reply]
> And these are the available controlnet for FLUX that I know
> 
> Xlabs v3 controlnets: https://huggingface.co/XLabs-AI/flux-controlnet-collections InstantX + Shakkerlabs union pro controlnet: https://huggingface.co/Shakker-Labs/FLUX.1-dev-ControlNet-Union-Pro Mistoline: https://huggingface.co/TheMistoAI/MistoLine_Flux.dev

But it has support now or not yet?

**Reply by nitinmukesh** [reply]
I don't know. I just mentioned the ones I found for Flux. Are these supported in Forge is only Devs can confirm.

**Comment by nitinmukesh** [comment]
Implementation is WiP as per the comment in following thread
https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/1685

**Comment by JohnneyTexas** [comment]
Line art (Es. realistic)
Depth (Es. dsine)
Normal map

**Comment by ZeroCool22** [comment]
We have support now?

**Reply by imago10** [reply]
Still nothing it will be so cool if we can please have some estimation is it gonna be after 1 week from now, month or never?

**Comment by GabrielSampol** [comment]
For me this is the only important thing missing.  Still no news on when will Forge support Controlnet?  Other platfoms like ComfyUI are already supporting it, but I wouldn't like to change to them unless it is really necessary.

**Reply by nitinmukesh** [reply]
I am also surprised. Earlier they were updating frequently but now they are just not updating.

**Comment by gabrielxcom** [comment]
_Are we there yet?_

**Reply by ZeroCool22** [reply]
> _Are we there yet?_

![Beast](https://github.com/user-attachments/assets/f6a4f0a5-7d88-4a10-b535-173c4c48255b)

**Comment by SunGreen777** [comment]
Perhaps work is underway on the internal part, which we cannot evaluate now, I really look forward to it and believe it

**Comment by juspar** [comment]
Following this. Really looking forward to controlnet for flux in Forge, don't wanna learn Comfy

**Reply by gabrielxcom** [reply]
> Following this. Really looking forward to controlnet for flux in Forge, don't wanna learn Comfy

I'm on the same boat, Comfy nodes and workflow look like NASA stuff

**Reply by imago10** [reply]
I learned Comfy just for this. And for some reason it is about 10x slower and the results look even worse than SD 1.5 with Controlnet in Forge. I asked people on forums to share their workflows. I got a lot of bugs, spent many days fixing them and sometimes I randomly got some strange bug during iterations. So for me it is not worth it...

**Reply by mperez96** [reply]
I agree ControlNet in ComfyUI is not working propertly, is a hit or miss, in some cases it works fine, in others it doesn't. So I am not so sure, but I believe the reason it is still not in Forge is because there is not even a way to make it work better in Comfy, is a wip still.

**Reply by juspar** [reply]
Note from [https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/932#discussion-7014318](https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/932#discussion-7014318) Flux Controlnet rewrite start has been delayed several times and is currently not slated to **start** until Nov 20

**Comment by gabrielxcom** [comment]
_**Are we there yet?**_ 👀

**Comment by mondermurshed** [comment]
Still nothing ?

**Comment by garbageplaidskids** [comment]
you all talking to a wall. lllyasviel left the development for a while. Forge is a dead project at this point. You guys should try to use comfy ui and get used to it. It also supports Hunyuan and Wan ai which you can generate texts to video and image to video. When forge going to support Hunyuan and Wan too? probably never lmao.

**Reply by imago10** [reply]
Well I tried ComfyUI and its super slow compared to Forge Flux also Controlnet works in strange way as a LORA (and its still super hard to get a control of an image) on my 16GB VRAM and resoults are even worse than in SD 1.5 or SDXL in Forge. And finding the right workflow and nodes that actually work takes almost a few days of searching - at that point I still dont have anything usable for Flux + Lora + Depth Controlnet. So still not worth it in my opinion... And waiting for Flux Controlnet in Forge or Automatic11111 ;)

**Reply by garbageplaidskids** [reply]
I understand but honestly it will probably take forever for them to add it. Also Flux isn't really that good, i think it's overrated. I can get a very nice results in SDXL that would look equivalent to Flux gens, except texts. Only problem in SDXL is texts imo.

**Reply by imago10** [reply]
Right - I just have a problem with training a good LORA for the character, maybe you can recommend me something? I tried KohyaSS but im not good at setting correct parameters and default presets give me bad results.

**Reply by garbageplaidskids** [reply]
Oh boy you are asking a wrong person lol. I am also looking for recommendations for it. I don't use lora training locally, i am using civitai's lora training. I just go with the default settings tbh, only changing batch size from 5 to 4 and increasing num repeats and epochs a little bit when my training dataset is low that's it.

**Reply by mondermurshed** [reply]
I hate Comfy UI fans, they are annoying af. 
WE DON'T WANT TO USE YOUR Comfy UI! SO PLEASE STOP RUIN OUR DEVELOPERS!

**Comment by wzwow** [comment]
is chroma working in forge now?
